charts/k8s-monitoring/templates/agent_config/_processors.river.txt

```diff
@@ -0,0 +1,194 @@
+    metrics = [otelcol.processor.filter.default.input]
+{{- end }}
+{{- if .Values.logs.enabled }}
+    logs = [otelcol.processor.filter.default.input]
+{{- end }}
+{{- if .Values.traces.enabled }}
+    traces = [otelcol.processor.filter.default.input]
+{{- end }}
+  }
+}
+
+otelcol.processor.filter "default" {
+  error_mode = "ignore"
+
+{{- if and .Values.metrics.enabled (or .Values.metrics.receiver.filters.metric .Values.metrics.receiver.filters.datapoint) }}
+  metrics = {
+{{- if .Values.metrics.receiver.filters.metric }}
+    metric = [
+{{- range $filter := .Values.metrics.receiver.filters.metric }}
+  {{ $filter | indent 6 }}
+{{- end }}
+    ]
+{{- end }}
+{{- if .Values.metrics.receiver.filters.datapoint }}
+    datapoint = [
+{{- range $filter := .Values.metrics.receiver.filters.datapoint }}
+  {{ $filter | indent 6 }}
+{{- end }}
+    ]
+{{- end }}
+  }
+{{- end }}{{- if and .Values.logs.enabled .Values.logs.receiver.filters.log_record }}
+  logs = {
+    log_record = [
+{{- range $filter := .Values.logs.receiver.filters.log_record }}
+  {{ $filter | indent 6 }}
+{{- end }}
+    ]
+{{- end }}
+{{- if and .Values.traces.enabled (or .Values.traces.receiver.filters.span .Values.traces.receiver.filters.spanevent) }}
+  traces = {
+{{- if .Values.traces.receiver.filters.span }}
+    span = [
+{{- range $filter := .Values.traces.receiver.filters.span }}
+  {{ $filter | indent 6 }}
+{{- end }}
+    ]
+{{- end }}
+{{- if .Values.traces.receiver.filters.spanevent }}
+    spanevent = [
+{{- range $filter := .Values.traces.receiver.filters.spanevent }}
+  {{ $filter | indent 6 }}
+{{- end }}
+    ]
+{{- end }}
+  }
+{{- end }}
+
+  output {
+{{- if .Values.metrics.enabled }}
+    metrics = [otelcol.processor.batch.batch_processor.input]
+{{- end }}
+{{- if .Values.logs.enabled }}
+    logs = [otelcol.processor.batch.batch_processor.input]
+{{- end }}
+{{- if .Values.traces.enabled }}
+    traces = [otelcol.processor.batch.batch_processor.input]
+{{- end }}
+  }
+}
+
+otelcol.processor.batch "batch_processor" {
+{{- with .Values.receivers.processors.batch }}
+  send_batch_size = {{ .size | int }}
+  send_batch_max_size = {{ .maxSize | int }}
+  timeout = {{ .timeout | quote}}
+{{- end }}
+  output {
+{{- if .Values.metrics.enabled }}
+    metrics = [otelcol.exporter.prometheus.metrics_converter.input]
+{{- end }}
+{{- if .Values.logs.enabled }}
+    logs = [otelcol.exporter.loki.logs_converter.input]
+{{- end }}
+{{- if .Values.traces.enabled }}
+{{- if eq .Values.externalServices.tempo.protocol "otlp" }}
+    traces = [otelcol.exporter.otlp.traces_service.input]
+{{- else if eq .Values.externalServices.tempo.protocol "otlphttp" }}
+    traces = [otelcol.exporter.otlphttp.traces_service.input]
+{{- end }}
+{{- end }}
+  }
+}
+{{- end }}
+
+{{- if .Values.metrics.enabled }}
+otelcol.exporter.prometheus "metrics_converter" {
+  forward_to = [prometheus.relabel.metrics_service.receiver]
+}
+{{- end }}
+
+{{- if .Values.logs.enabled }}
```

Comment:
If logs are enabled but pod logs are disabled, this snippet results in an error. E.g. helm values snippet like
```
logs:
  enabled: true
  cluster_events:
    enabled: true
  pod_logs:
    enabled: false
```